lib/ex_unit_properties.ex

```diff
@@ -222,7 +222,11 @@ defmodule ExUnitProperties do
     * value generation clauses affect shrinking similarly to `bind/2`
 
   """
-  # TODO: support do: syntax
+  defmacro gen({:all, _meta, clauses_with_body}) do
+    [[do: body] | clauses] = Enum.reverse(clauses_with_body)
```

Comment:
Don't we need to `Enum.reverse(clauses)` to bring the clauses back to the original order here as well?

Reply:
My understanding is, that there are no options in the `gen all` macro and there are only generation clauses for types of data. And the order you give the clauses is the reverse of the order the variables are injected through `var!` into the body's binding. Reversing them here would only cause difference if for some reason you have this:
```
      data = gen all int <- binary(),
                             int <- integer(), do: int

      check all int <- data do
        assert is_integer(int)
      end
```
So this would fail, since we have reversed the order of the clauses and the second integer is actually injected first and being overwritten by the first one. I can do a second reverse, so that there is minimum surprise there?

Reply:
Subsequent clauses can depend on previous clauses:

```elixir
data = gen all list <- list_of(integer()),
               member <- member_of(list),
               do: {list, member}
```

This would not work without reversing the clauses back to the original order.

Reply:
Oh, right :sweat_smile: , anyway, added it, should I add some tests for that too?